xbmc/network/GUIDialogNetworkSetup.cpp

```diff
@@ -371,94 +366,69 @@ void CGUIDialogNetworkSetup::UpdateButtons()
-    url.SetProtocol("ftp");
-  else if (m_protocol == NET_PROTOCOL_HTTP)
-    url.SetProtocol("http");
-  else if (m_protocol == NET_PROTOCOL_HTTPS)
-    url.SetProtocol("https");
-  else if (m_protocol == NET_PROTOCOL_DAV)
-    url.SetProtocol("dav");
-  else if (m_protocol == NET_PROTOCOL_DAVS)
-    url.SetProtocol("davs");
-  else if (m_protocol == NET_PROTOCOL_UPNP)
-    url.SetProtocol("upnp");
-  else if (m_protocol == NET_PROTOCOL_RSS)
-    url.SetProtocol("rss");
-  else if (m_protocol == NET_PROTOCOL_NFS)
-    url.SetProtocol("nfs");
-  else if (m_protocol == NET_PROTOCOL_SFTP)
-    url.SetProtocol("sftp");
-    
+  url.SetProtocol(m_protocols[m_protocol].type);
+
   if (!m_username.empty())
   {
     url.SetUserName(m_username);
     if (!m_password.empty())
       url.SetPassword(m_password);
   }
-  if(!m_server.empty())
+
+  if (!m_server.empty())
     url.SetHostName(m_server);
-  if (((m_protocol == NET_PROTOCOL_FTP) ||
-       (m_protocol == NET_PROTOCOL_HTTP) ||
-       (m_protocol == NET_PROTOCOL_HTTPS) ||
-       (m_protocol == NET_PROTOCOL_DAV) ||
-       (m_protocol == NET_PROTOCOL_DAVS) ||
-       (m_protocol == NET_PROTOCOL_RSS) ||
-       (m_protocol == NET_PROTOCOL_SFTP) ||
-       (m_protocol == NET_PROTOCOL_NFS))
-      && !m_port.empty() && atoi(m_port.c_str()) > 0)
+
+  if (m_protocols[m_protocol].supportPort &&
+      !m_port.empty() && atoi(m_port.c_str()) > 0)
   {
     url.SetPort(atoi(m_port.c_str()));
   }
+
   if (!m_path.empty())
     url.SetFileName(m_path);
+
   return url.Get();
 }
 
-void CGUIDialogNetworkSetup::SetPath(const std::string &path)
+bool CGUIDialogNetworkSetup::SetPath(const std::string &path)
 {
   CURL url(path);
-  if (url.IsProtocol("smb"))
-    m_protocol = NET_PROTOCOL_SMB;
-  else if (url.IsProtocol("ftp"))
-    m_protocol = NET_PROTOCOL_FTP;
-  else if (url.IsProtocol("http"))
-    m_protocol = NET_PROTOCOL_HTTP;
-  else if (url.IsProtocol("https"))
-    m_protocol = NET_PROTOCOL_HTTPS;
-  else if (url.IsProtocol("dav"))
-    m_protocol = NET_PROTOCOL_DAV;
-  else if (url.IsProtocol("davs"))
-    m_protocol = NET_PROTOCOL_DAVS;
-  else if (url.IsProtocol("upnp"))
-    m_protocol = NET_PROTOCOL_UPNP;
-  else if (url.IsProtocol("rss"))
-    m_protocol = NET_PROTOCOL_RSS;
-  else if (url.IsProtocol("nfs"))
-    m_protocol = NET_PROTOCOL_NFS;
-  else if (url.IsProtocol("sftp") || url.IsProtocol("ssh"))
-    m_protocol = NET_PROTOCOL_SFTP;
-  else
-    m_protocol = NET_PROTOCOL_SMB;  // default to smb
+  m_protocol = -1;
+  for (size_t i = 0; i < m_protocols.size(); ++i)
+  {
+    if (m_protocols[i].type == url.GetProtocol())
+    {
+      m_protocol = i;
+      break;
+    }
+  }
+  if (m_protocol == -1)
+  {
+    if (!path.empty())
+      CLog::Log(LOGERROR, "__PRETTY_FUNCTION__: Asked to initialize for unknown path %s", path.c_str());
+    m_username.clear();
+    m_password.clear();
+    m_port.clear();
+    m_server.clear();
+    m_path.clear();
+    m_protocol = 0;
```

Comment:
we could add a new method `Reset()` like we did in other dialogs IIRC and call it here and in case the path is empty. This way we could call it earlier and do not need to loop over protocols etc.

Reply:
the nanosecs that takes we will lose a million time over having this discussion ;)

Reply:
sure sure, but it's handy if we need to somewhere else. You could leave it as it is if it's worth the time

Reply:
done.